src/openms/include/OpenMS/FORMAT/FASTAFile.h

```diff
@@ -38,187 +38,175 @@
+            {
+                return sequence == rhs.sequence;
+            }
+        };
+
+        /// Default constructor
+        FASTAFile();
+
+        /// Destructor
+        virtual ~FASTAFile();
+
+        /**
+          @brief Prepares a FASTA file given by 'filename' for streamed reading using readNext().
+          @exception Exception::FileNotFound is thrown if the file does not exists.
+          @exception Exception::ParseError is thrown if the file does not suit to the standard.
+        */
+        void readStart(const String& filename);
+
+        /**
+        @brief Reads the next FASTA entry from file.
+        If you want to read all entries in one go, use load().
+        @return true if entry was read; false if eof was reached
+        @exception Exception::FileNotFound is thrown if the file does not exists.
+        @exception Exception::ParseError is thrown if the file does not suit to the standard.
+        */
+        bool readNext(FASTAEntry& protein);
+
+        /// current stream position
+        std::streampos position();
+
+        /// is stream at EOF?
+        bool atEnd();
+
+        /// seek stream to @p pos
+        bool setPosition(const std::streampos& pos);
+
+        /**
+        @brief Prepares a FASTA file given by 'filename' for streamed writing using writeNext().
+        @exception Exception::UnableToCreateFile is thrown if the process is not able to write to the file (disk full?).
+        */
+        void writeStart(const String& filename);
+
+        /**
+        @brief Stores the data given by @p protein. Call writeStart() once before calling writeNext().
+        Call writeEnd() when done to close the file!
+        @exception Exception::UnableToCreateFile is thrown if the process is not able to write the file.
+        */
+        void writeNext(const FASTAEntry& protein);
+
+        /**
+        @brief Closes the file (flush). Called implicitly when FASTAFile object does out of scope.
+        */
+        void writeEnd();
+
+
+        /**
+          @brief loads a FASTA file given by 'filename' and stores the information in 'data'
+          This uses more RAM than readStart() and readNext().
+          @exception Exception::FileNotFound is thrown if the file does not exists.
+          @exception Exception::ParseError is thrown if the file does not suit to the standard.
+        */
+        void static load(const String& filename, std::vector<FASTAEntry>& data);
+
+        /**
+            @brief stores the data given by 'data' at the file 'filename'
+
+            This uses more RAM than writeStart() and writeNext().
+            @exception Exception::UnableToCreateFile is thrown if the process is not able to write the file.
+          */
+        void static store(const String& filename, const std::vector<FASTAEntry>& data);
+
+    protected:
+        bool readEntry_(std::string & id, std::string & seq);///<  reading a protein entry and saving the ID and
```

Comment:
for multi-line comments prefer (also note the formatting `string& id` instead of `string & id`. (see CodingConventions)
```
/**
  @brief Reads a protein entry from the current file position and returns the ID and sequence
*/
bool readEntry_(std::string& id, std::string& seq);
```